privacyidea/models.py

```diff
@@ -382,7 +393,9 @@ def set_pin(self, pin, hashed=True):
         if hashed is True:
             self.set_hashed_pin(upin)
             log.debug("setPin(HASH:{0!r})".format(self.pin_hash))
-        elif hashed is False:
+        else:
+            if isinstance(pin, six.text_type):
+                upin = upin.encode('utf8')
```

Comment:
Correct me if I'm wrong, but can't we remove this because ``encryptPin`` will call ``_to_bytes()`` anyway?

Reply:
No need for correction :-)

Reply:
fixed in d709698

Reply:
:+1: